tests/unit/modules/test_cassandra_cql.py

```diff
@@ -79,3 +79,19 @@ def test_unspecified_opts(self):
         with patch.dict(cassandra_cql.__salt__, {'config.option': MagicMock(return_value={})}):
             self.assertEqual(cassandra_cql._get_ssl_opts(),  # pylint: disable=protected-access
                              None)
+
+    def test_valid_async_args(self):
+        mock_execute = MagicMock(return_value={})
+        mock_execute_async = MagicMock(return_value={})
+        mock_context = {
+            'cassandra_cql_returner_cluster': MagicMock(return_value={}),
+            'cassandra_cql_returner_session': MagicMock(execute=mock_execute,
+                                                        execute_async=mock_execute_async,
+                                                        prepare=lambda _: MagicMock(bind=lambda _: None),  # mock prepared_statement
+                                                        row_factory=None),
+            'cassandra_cql_prepared': {}
+        }
+
+        with patch.dict(cassandra_cql.__context__, mock_context):
+            cassandra_cql.cql_query_with_prepare('SELECT now() from system.local;', 'select_now', [], asynchronous=True)
```

Comment:
can we also ensure `async` works as well here?

Reply:
I add test for ensuring `async` works. 
but since python 3.7, async is reserved keyword so I have no choice but to pass async param as dict.